File: src/18.ts
import fs from 'fs'

const source = fs.readFileSync('./res/popular-names.txt', 'utf-8')
const compare = (a: number, b: number) => {
    if (a < b) return -1
    if (a > b) return +1
    return 0
}
const pickNumber = (row: string) =>
    Number(row.split('\t')[2])
console.log(
    source
    .split('\n')
    .sort((a, b) => compare(pickNumber(b), pickNumber(a)))
    .join('\n')
)

File: src/18.py
# sort -t$'\t' -k3nr res/popular-names.txt

from .helper.file_helper import read_string

source = read_string("./res/popular-names.txt")

print("\n".join(sorted(source.split("\n"), key=lambda row: int(row.split("\t")[2]), reverse=True,)))
